api/server/router/network/filter.go

```diff
@@ -93,18 +48,40 @@ func filterNetworks(nws []libnetwork.Network, filter filters.Args) ([]libnetwork
 	}
 
 	var displayNet []libnetwork.Network
-	for fkey, fhandler := range supportedFilters {
-		errFilter := filter.WalkValues(fkey, func(fval string) error {
-			passList, err := fhandler(nws, fval)
+	for _, nw := range nws {
+		if filter.Include("name") {
+			if !filter.Match("name", nw.Name()) {
+				continue
+			}
+		}
+		if filter.Include("id") {
+			if !filter.Match("id", nw.ID()) {
+				continue
+			}
+		}
+		if filter.Include("label") {
+			if !filter.MatchKVList("label", nw.Info().Labels()) {
+				continue
+			}
+		}
+		displayNet = append(displayNet, nw)
+	}
+
+	if filter.Include("type") {
+		var typeNet []libnetwork.Network
+		errFilter := filter.WalkValues("type", func(fval string) error {
```

Comment:
hi @thaJeztah The reason we use that, is 
1) filterNetworkByType is specific logic in network, general filter(engine-api/types/filters/parse.go) not include that, and we'd better not put such logic in general filters

2) we directly use filter.WalkValues, which I can reduce duplicate filter logic, if we want to above loop ways, we need write another function, which fetch filter "type" values, and validate it and then return bool etc. I think is not good idea.

Please let me know what you think.  Thanks